app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx

```diff
@@ -210,8 +210,17 @@ describe('<ArticleCoverImage />', () => {
         );
 
         // Fire a change event in the hidden input with JSON payload for success
-        const fakeSuccessMessage = { 'action': 'success', 'link': '/some-fake-image.jpg', 'namespace': 'coverUpload' };
-        const event = createEvent('ForemMobile', document, { detail: fakeSuccessMessage }, { EventType: 'CustomEvent' });
+        const fakeSuccessMessage = JSON.stringify({
```

Comment:
These tests were failing because JSON was expected, not a JS object.

Reply:
The spec is still failing but for a reason I don't understand. I'm running it using this command:

```bash
npx jest app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx
```

I changed the names and added this debug log to `ArticleCoverImage.jsx`

```js
console.log("Is this undefined::::: ", onMainImageUrlChange);
```

And I see this ([travis output here](https://app.travis-ci.com/github/forem/forem/jobs/548908065#L973)):

<img width="1311" alt="Screen Shot 2021-11-19 at 09 42 32" src="https://user-images.githubusercontent.com/6045239/142651349-fa0b0909-722f-4950-b0f3-6a4664061fef.png">

But the `onMainImageUrlChangeSpy` was just set a few lines above here as `jest.fn();`. 

@nickytonline any ideas why it's arriving there as `undefined`?

Reply:
The problem is when you run `this.props`, this is the Document (window), and not the Preact component.

![image](https://user-images.githubusercontent.com/833231/142677252-8ff44919-6c1a-4046-94cd-0f4d87ac3e34.png)

I'm not sure why that is happening as that function is using an arrow function so it should keep the `this` as the Preact component. Using `bind` seems to fix it, but I'm going to dig a bit more.

`document.addEventListener('ForemMobile', this.handleNativeMessage.bind(this));`

Reply:
Hmm, digging a bit deeper and that does not appear to be the issue as the test passes in isolation.

![CleanShot 2021-11-19 at 15 10 23](https://user-images.githubusercontent.com/833231/142685618-f9ff4ac0-3a12-4ae9-9e2c-5fecdfa0657b.png)

Reply:
OK, I've narrowed it down. If I skip this test, it's all green. Getting closer.

```jsx
    it.skip('triggers a webkit messageHandler call when isNativeIOS', async () => {
      global.window.ForemMobile = { injectNativeMessage: jest.fn() };

      const { queryByLabelText } = render(<ArticleCoverImage mainImage="" />);
      const uploadButton = queryByLabelText(/Upload cover image/i);
      uploadButton.click();
      expect(
        global.window.ForemMobile.injectNativeMessage,
      ).toHaveBeenCalledTimes(1);
    });
```

Reply:
Alright. It's all fixed. A missing prop was the culprit.

```diff
- const { queryByLabelText } = render(<ArticleCoverImage mainImage="" />);
+ const { queryByLabelText } = render(
+ <ArticleCoverImage mainImage="" onMainImageUrlChange={jest.fn()} />,
+);
```